### PR TITLE
[camera]handle iOS camera access permission

### PR DESCRIPTION
**Pull request by hellohuanlin**
This PR adds camera access permission handling logic in iOS camera plugin. This fixes a red screen crash when first time using the camera. 

In the future we should clean up all errors. Ideally there should be a header with a list of all errors, so that it's easier to refer to. But to limit the scope of this PR, I only added permission errors following existing conventions. 

## Links

Issue [here](https://github.com/flutter/flutter/issues/97199)

Proposal [here](https://docs.google.com/document/d/1t_5FYzOTlBWC6arHc4_8VuG0BUOtIyP9xWvbDER5Y9I/edit)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/master/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/camera/camera/ios/Classes/CameraPermissionUtils.m
Comment (jmagman) [inline packages/camera/camera/ios/Classes/CameraPermissionUtils.m:15]: Do we need to add documentation to the README for these errors?

https://github.com/flutter/plugins/blob/08e4e02ef2a104e591fcaa92f8ff1f57775f75af/packages/camera/camera/README.md?plain=1#L79

What about the example app?
https://github.com/flutter/plugins/blob/08e4e02ef2a104e591fcaa92f8ff1f57775f75af/packages/camera/camera/example/lib/main.dart

**Review by jmagman — COMMENTED** [review]
Once @cyanglaz finishes https://github.com/flutter/flutter/issues/93504 you should add an `XCUITest` that validates the right thing happens when permission is granted/denied.

**Comment by jmagman** [comment]
> Once @cyanglaz finishes [flutter/flutter#93504](https://github.com/flutter/flutter/issues/93504) you should add an `XCUITest` that validates the right thing happens when permission is granted/denied.

For the record that work does not need to block this PR.

File: packages/camera/camera/ios/Classes/CameraPermissionUtils.m
Comment (hellohuanlin) [reply packages/camera/camera/ios/Classes/CameraPermissionUtils.m:15]: Updated example and README. Also checked that the short code snippet in the README file still works.

File: packages/camera/camera/README.md
Comment (AlexV525) [inline packages/camera/camera/README.md:101]: This seems not consistent with other exceptions cases? Is this intentional?

Comment (hellohuanlin) [reply packages/camera/camera/README.md:101]: This is a legacy code for Android and Web that we will remove soon. I still include it because with this current version we will need to handle them.

File: packages/camera/camera/README.md
Comment (jmagman) [inline packages/camera/camera/README.md:144]:
```
Running command: "dart run code_excerpt_updater --write-in-place --yaml --no-escape-ng-interpolation ../README.md" in /tmp/cirrus-ci-build/packages/camera/camera/example
Processed 1 Dart/Jade/Markdown files: 1 out of 2 fragments needed updating.
�[31mREADME.md is out of sync with its source excerpts.

If you edited code in README.md directly, you should instead edit the example source files. If you edited source files, run the repository tooling's "update-excerpts" command on this package, and update your PR with the resulting changes.
```
Check out https://github.com/flutter/plugins/pull/5339/files#diff-6c463281036ceb5ca6aeb83f1e11574c52ea1a063c88613afa558454b64a943f for example

**Review by jmagman — COMMENTED** [review]
LGTM once the README code excerpt checker is happy.

File: packages/camera/camera/README.md
Comment (hellohuanlin) [reply packages/camera/camera/README.md:144]: Good to know! Updated.

Comment (jmagman) [reply packages/camera/camera/README.md:144]:
```
  error - example/lib/readme_full_example.dart:39:20 - Missing parameter type for 'e'. Try adding an explicit type, or remove implicit-dynamic from your analysis options file. - implicit_dynamic_parameter
   info - example/lib/readme_full_example.dart:39:20 - Specify type annotations. - always_specify_types

2 issues found.
```

File: packages/camera/camera/CHANGELOG.md
Comment (stuartmorgan) [inline packages/camera/camera/CHANGELOG.md:3]: "when using the camera for the first time"

File: packages/camera/camera/CHANGELOG.md
Comment (stuartmorgan) [inline packages/camera/camera/CHANGELOG.md:29]: We don't generally change already-published changelog entries, even if they had style issues.

File: packages/camera/camera/README.md
Comment (stuartmorgan) [inline packages/camera/camera/README.md:89]: Please use backticks for these error codes to highlight that they are code.

File: packages/camera/camera/README.md
Comment (stuartmorgan) [inline packages/camera/camera/README.md:90]: denies

Comment (stuartmorgan) [inline packages/camera/camera/README.md:90]: The newlines here aren't doing anything here in the output (see alternate view). You should instead format these as a single paragraph (which is what they render as), with at `:` after the code.

File: packages/camera/camera/README.md
Comment (stuartmorgan) [reply packages/camera/camera/README.md:144]: code-excerpt for the win! :)

File: packages/camera/camera/example/lib/main.dart
Comment (stuartmorgan) [inline packages/camera/camera/example/lib/main.dart:703]: How about `'Unknown permission error.`'

File: packages/camera/camera/ios/Classes/CameraPermissionUtils.h
Comment (stuartmorgan) [inline packages/camera/camera/ios/Classes/CameraPermissionUtils.h:10]: Blank comment line after this to make a paragraph please (similar to [Dart style](https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph))

File: packages/camera/camera/ios/Classes/CameraPermissionUtils.h
Comment (stuartmorgan) [inline packages/camera/camera/ios/Classes/CameraPermissionUtils.h:14]: Blank line above this too.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/camera/camera/README.md
Comment (stuartmorgan) [inline packages/camera/camera/README.md:136]: Please don't use `dynamic` in modern Dart except where absolutely necessary; it bypasses all type checks. `Object` (or when needed, `Object?`) is almost always the better choice.

**Review by stuartmorgan — APPROVED** [review]
LGTM with one small change.

**Review by jmagman — APPROVED** [review]
LGTM

**Pull request merged**
